### PR TITLE
TW-397: change ux when send first message in draft chat

### PR DESCRIPTION
**Pull request by sherlockvn**
### Demo device:
https://github.com/linagora/twake-on-matrix/assets/43041967/468eee0b-d441-440d-8521-d1280431a24b

### Demo web:

https://github.com/linagora/twake-on-matrix/assets/43041967/e5fb873b-f222-4df6-a3bd-c4283880d5c8

### CODE REVIEW

**Comment by nqhhdev** [comment]
1. WHAT IS HAPPEN IN WEB VERSION? :)

File: lib/pages/chat_draft/draft_chat.dart [hunk 1]
```diff
@@ -157,20 +159,23 @@ class DraftChatController extends State<DraftChat>
...
-          );
+          // showDialog(
+          //   useRootNavigator: false,
+          //   context: context,
+          //   builder: (context) {
+          //     return const Center(
+          //       child: CircularProgressIndicator(),
+          //     );
+          //   },
+          // );
```
Comment (drminh2807) [inline lib/pages/chat_draft/draft_chat.dart:170]: remove it completely

File: lib/pages/chat_draft/draft_chat.dart [hunk 2]
```diff
@@ -157,20 +159,23 @@ class DraftChatController extends State<DraftChat>
...
         } else if (success is CreateDirectChatSuccess) {
           final room = Matrix.of(context).client.getRoomById(success.roomId);
           if (room != null) {
             onRoomCreatedSuccess?.call(room);
-            context.go('/rooms/${room.id}/');
+            context.go(
+              '/rooms/${room.id}/',
+              extra: 'fromDraftChat',
```
Comment (nqhhdev) [inline lib/pages/chat_draft/draft_chat.dart:177]: IT'S BAD FOR ME. WANT STRONG TYPE ?

Comment (nqhhdev) [reply lib/pages/chat_draft/draft_chat.dart:177]: 'fromDraftChat' ????? what is this, it harrd for me

File: lib/pages/chat_draft/draft_chat.dart (hunk 1 above)
Comment (nqhhdev) [inline lib/pages/chat_draft/draft_chat.dart:170]: WHy don't delete it?

File: lib/config/go_routes/go_router.dart [hunk 3]
```diff
@@ -357,13 +357,20 @@ abstract class AppRoutes {
...
               pageBuilder: (context, state) {
-                final shareFile = state.extra as MatrixFile?;
+                if (state.extra == 'fromDraftChat') {
+                  return NoTransitionPage(
+                    child: Chat(
+                      roomId: state.pathParameters['roomid']!,
+                      key: Key(state.pathParameters['roomid']!),
+                    ),
+                  );
+                }
```
Comment (nqhhdev) [inline lib/config/go_routes/go_router.dart:367]: WHEN USE BACK ON WEB BROWSER WHAT IS HAPPEN ?

Comment (nqhhdev) [reply lib/config/go_routes/go_router.dart:367]: VERRRYYY BADD  `'fromDraftChat'`.

File: lib/pages/chat_draft/draft_chat.dart [hunk 4]
```diff
@@ -58,6 +57,8 @@ class DraftChatController extends State<DraftChat>
 
   EmojiPickerType emojiPickerType = EmojiPickerType.keyboard;
 
+  final isSendingNotifier = ValueNotifier(false);
```
Comment (nqhhdev) [inline lib/pages/chat_draft/draft_chat.dart:60]:
```suggestion
  final ValueNotifier<bool> isSendingNotifier = ValueNotifier(false);
```

STRONG STRONGGGGGG

Comment (sherlockvn) [reply lib/pages/chat_draft/draft_chat.dart:60]: no need, because it's clear in the right hand side

Comment (nqhhdev) [reply lib/pages/chat_draft/draft_chat.dart:60]: !

Screenshot 2023-08-30 at 16 18 13

File: lib/config/go_routes/go_router.dart (hunk 3 above)
Comment (sherlockvn) [reply lib/config/go_routes/go_router.dart:367]: check the demo

File: lib/pages/chat_draft/draft_chat.dart (hunk 4 above)
Comment (sherlockvn) [reply lib/pages/chat_draft/draft_chat.dart:60]: Screenshot 2023-08-30 at 16 25 31

Comment (drminh2807) [reply lib/pages/chat_draft/draft_chat.dart:60]: agree with Khải

File: lib/pages/chat_draft/draft_chat.dart [hunk 5]
```diff
@@ -154,23 +156,18 @@ class DraftChatController extends State<DraftChat>
...
-            useRootNavigator: false,
-            context: context,
-            builder: (context) {
-              return const Center(
-                child: CircularProgressIndicator(),
-              );
-            },
-          );
-        } else if (success is CreateDirectChatSuccess) {
+        if (success is CreateDirectChatSuccess) {
```
[the remark this answers is not in the document]
Comment (sherlockvn) [reply lib/pages/chat_draft/draft_chat.dart:161]: later other cases are needed to handled, I have to change it back

**Comment by sherlockvn** [comment]
> image
> IMO, it seems not good, TextInput is expanding, please try to keep the position and the size for all things, only change from send button to progress indicator


https://github.com/linagora/twake-on-matrix/assets/43041967/64aaf39e-023a-4522-8f9f-086e5432cc96

**Review by hoangdat — APPROVED** [review]

**Review by drminh2807 — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**